src/services/transactionService.ts

```diff
@@ -349,92 +374,158 @@ export const buildPaymentTransaction = async (
       networkPassphrase: networkDetails.networkPassphrase,
     });
 
-    if (memo) {
+    const isCustomToken =
+      selectedBalance &&
+      "contractId" in selectedBalance &&
+      selectedBalance.contractId;
+
+    const isToContractAddress = isContractId(recipientAddress);
+    const shouldUseSorobanTransfer = isToContractAddress || isCustomToken;
+    const isRecipientMuxed = isMuxedAccount(recipientAddress);
+    if (memo && !shouldUseSorobanTransfer && !isRecipientMuxed) {
       transactionBuilder.addMemo(new Memo(Memo.text(memo).type, memo));
     }
 
-    const isToContractAddress = isContractId(recipientAddress);
+    if (shouldUseSorobanTransfer) {
+      let contractId: string;
+      let token: SdkToken;
+
+      if (isCustomToken && selectedBalance.contractId) {
```

Comment:
```suggestion
      if (isCustomToken) {
```

isCustomToken already implies selectedBalance.contractId exists, is lint complaining here?

Reply:
not really. leftover from initial development